File: app/src/main/java/com/nicholasdoglio/eyebleach/util/Intents.java
/*
    Aww Gallery
    Copyright (C) 2017  Nicholas Doglio

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.nicholasdoglio.eyebleach.util;

import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.net.Uri;
import android.view.View;

/**
 * @author Nicholas Doglio
 */
public class Intents {

    public void openWebPage(Context context, String url) {
        Uri webpage = Uri.parse(url);
        Intent webIntent = new Intent(Intent.ACTION_VIEW, webpage);
        if (webIntent.resolveActivity(context.getPackageManager()) != null) {
            context.startActivity(webIntent);
        }
    }

    public void shareUrl(Context context, String url) {
        Intent shareIntent = new Intent(Intent.ACTION_SEND);
        shareIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
        shareIntent.putExtra(Intent.EXTRA_TEXT, url);
        shareIntent.setType("text/plain");
        context.startActivity(Intent.createChooser(shareIntent, "Share your cute animals via: "));
    }

    public void composeEmail(Context context) {
        Intent emailIntent = new Intent(Intent.ACTION_SENDTO);
        emailIntent.setData(Uri.parse("mailto: NicholasDoglio@Gmail.com"));
        emailIntent.putExtra(Intent.EXTRA_SUBJECT, "Aww Gallery Feedback");
        if (emailIntent.resolveActivity(context.getPackageManager()) != null) {
            context.startActivity(emailIntent);
        }
    }

    public void startDetailActivity(View view, int position) {

    }

    public void startAboutActivity(Context context) {

    }

    public String provideVersion(Context context) {
        String version = "";

        try {
            PackageInfo packageInfo = context.getPackageManager().getPackageInfo(context.getPackageName(), 0);
            version = packageInfo.versionName;
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }
        return "Version: " + version;
    }
}

File: app/src/main/java/com/nicholasdoglio/eyebleach/util/Intents.kt
/*
 * MIT License
 *
 * Copyright (c) 2019 Nicholas Doglio
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.nicholasdoglio.eyebleach.util

import android.content.Context
import com.nicholasdoglio.core.BuildConfig

/**
 * @author Nicholas Doglio
 */
class Intents {

    // TODO move back to Anko? Move this into extension functions?

    fun openWebPage(context: Context, url: String) {
//        context.browse(url)
    }

    fun shareUrl(context: Context, url: String) {
// TODO look into ShareCompat
//        val shareIntent = ShareCompat.IntentBuilder.from(activity)
//            .setType("text/plain")
//            .setText(url)
//            .setChooserTitle("Share your cute animals via: ")
//            .intent
//        context.share(url,"Share your cute animals via: ")

//        val shareIntent = Intent(Intent.ACTION_SEND)
//        shareIntent.flags = Intent.FLAG_ACTIVITY_CLEAR_TASK
//        shareIntent.putExtra(Intent.EXTRA_TEXT, url)
//        shareIntent.type = "text/plain"
//        context.startActivity(Intent.createChooser(shareIntent, "Share your cute animals via: "))
    }

    fun composeEmail(context: Context) {
//        context.email("NicholasDoglio@Gmail.com", "Aww Gallery Feedback")
    }

    fun provideVersion(context: Context): String { // This is unnecessary, move later
        return "Version: ${BuildConfig.VERSION_NAME}"
    }
}